python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -295,8 +296,8 @@ def build_asset_ins(
         )
 
     for asset_key in non_argument_deps:
-        stringified_asset_key = asset_key.to_string(legacy=True)
+        stringified_asset_key = "_".join(asset_key.path)
```

Comment:
Why the change to this join? The param name doesn't need to exist in the asset definition signature, so it might be nice to use the `to_string` for consistency

Reply:
asset_key.to_string() returns (basically) `".".join(asset_key.path)`. so it actually will raise an error if you have an asset key with multiple segments to the key (i.e. `AssetKey(["foo", "bar"])`), because "foo.bar" is not a valid identifier (but "foo_bar" is).

Reply:
Oh, I see. thanks for fixing!